webhook.go

```diff
@@ -64,20 +75,340 @@ type Registration struct {
+	//Default value will be set to none
+	RetryEachUrl int `json:"retry_each_url"`
+
+	//MaxRetry is the total amount times a request will be retried.
+	MaxRetry int `json:"max_retry"`
+}
+
+// Webhook is a substructure with data related to event delivery.
+type Webhook struct {
+	// Accept is the encoding type of outgoing events. The following encoding types are supported, otherwise
+	// a 406 response code is returned: application/octet-stream, application/json, application/jsonl, application/msgpack.
+	// Note: An `Accept` of application/octet-stream or application/json will result in a single response for batch sizes of 0 or 1
+	// and batch sizes greater than 1 will result in a multipart response. An `Accept` of application/jsonl or application/msgpack
+	// will always result in a single response with a list of batched events for any batch size.
+	Accept string `json:"accept"`
+
+	// AcceptEncoding is the content type of outgoing events. The following content types are supported, otherwise
+	// a 406 response code is returned: gzip.
+	AcceptEncoding string `json:"accept_encoding"`
+
+	// Secret is the string value.
+	// (Optional, set to "" to disable behavior).
+	Secret string `json:"secret,omitempty"`
+
+	// SecretHash is the hash algorithm to be used. Only sha256 HMAC and sha512 HMAC are supported.
+	// (Optional).
+	// The Default value is the largest sha HMAC supported, sha512 HMAC.
+	SecretHash string `json:"secret_hash"`
+
+	// If true, response will use the device content-type and wrp payload as its body
+	// Otherwise, response will Accecpt as the content-type and wrp message as its body
+	// Default: False (the entire wrp message is sent)
+	PayloadOnly bool `json:"payload_only"`
+
+	// ReceiverUrls is the list of receiver urls that will be used where as if the first url fails,
+	// then the second url would be used and so on.
+	// Note: either `ReceiverURLs` or `DNSSrvRecord` must be used but not both.
+	ReceiverURLs []string `json:"receiver_urls"`
+
+	// DNSSrvRecord is the substructure for configuration related to load balancing.
+	// Note: either `ReceiverURLs` or `DNSSrvRecord` must be used but not both.
+	DNSSrvRecord struct {
+		// FQDNs is a list of FQDNs pointing to dns srv records
+		FQDNs []string `json:"fqdns"`
+
+		// LoadBalancingScheme is the scheme to use for load balancing. Either the
+		// srv record attribute `weight` or `priortiy` can be used.
+		LoadBalancingScheme string `json:"load_balancing_scheme"`
+	} `json:"dns_srv_record"`
+
+	//RetryHint is the substructure for configuration related to retrying requests.
+	// (Optional, if omited then retries will be based on default values defined by server)
+	RetryHint RetryHint `json:"retry_hint"`
+}
+
+// Kafka is a substructure with data related to event delivery.
+type Kafka struct {
+	// Accept is content type value to set WRP messages to (unless already specified in the WRP).
+	Accept string `json:"accept"`
```

Comment:
```suggestion
	// Accept is the encoding type of outgoing events. The following encoding types are supported, otherwise
	// a 406 response code is returned: application/octet-stream, application/json, application/jsonl, application/msgpack.
	// Note: An `Accept` of application/octet-stream or application/json will result in a single response for batch sizes of 0 or 1
	// and batch sizes greater than 1 will result in a multipart response. An `Accept` of application/jsonl or application/msgpack
	// will always result in a single response with a list of batched events for any batch size.
	Accept string `json:"accept"`
```